src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -552,6 +552,9 @@ def samples(self, new_samples):
         samples: np.ndarray
             The samples for the Fourier Operator.
         """
+        xp = get_array_module(new_samples)
+        if xp.__name__ == "cupy":
+            new_samples = new_samples.get()
```

Comment:
you can be more defensive with a `_array_to_numpy` from our `_array_compat`  module 

```suggestion
    new_samples = _array_to_numpy(new_samples)
```